feature(nimbus): Add deliveries table to feature health page.

b4handjr opened the pull request:
Because

- We want to display data about recent deliveries based on a feature/application combo on the feature health page

This commit

- Adds a deliveries table to the feature health page

Fixes #13485 #13610

b4handjr commented:
View no features
<img width="1978" height="425" alt="image" src="https://github.com/user-attachments/assets/35995aab-a02f-4589-88e3-c98b089134dc" />

b4handjr commented on experimenter/experimenter/nimbus_ui/templates/nimbus_experiments/features.html:
```diff
@@ -30,11 +30,47 @@
+                  {% else %}
+                    <td id="delivery-type-experiment" class="p-2 border-1">Experiment</td>
+                  {% endif %}
+                  <td id="delivery-channel" class="p-2 border-1">{{ experiment.get_channel_display }}</td>
+                  <td id="delivery-min-version" class="p-2 border-1">{{ experiment.firefox_min_version|parse_version }}</td>
+                  <td id="delivery-population-percent" class="p-2 border-1">{{ experiment.population_percent|floatformat:"-1" }}</td>
+                  <td id="delivery-brief" class="p-2 border-1">{{ experiment.delivery_brief|format_not_set }}</td>
```
So you just want the row defined with a border?

yashikakhurana reviewed (changes requested):
I think we need to remove borders, it is showing 2 borders now, lets say consistent with the design of tables with the home page. Also noticed 2 bugs
1: Look how it is showing the selected feature, if I select that feature again it is not showing any deliveries

https://github.com/user-attachments/assets/7f9610ea-f93a-4b4f-a508-bceaab0fc7a1

2: If you select the different application, deliveries data is not getting changed, it is still showing the old data

https://github.com/user-attachments/assets/bf46e67c-f4f3-4050-9776-6938aad6bcc1

b4handjr commented:
How does this look
<img width="2002" height="789" alt="image" src="https://github.com/user-attachments/assets/1f9ea3b6-73f8-4dba-be2d-665400d00084" />

I also fixed the mix of deliveries that was showing up when you first navigated to the page

yashikakhurana commented:
@b4handjr It is looking really amazing, some minor suggestions would be like to add the same kind of background where we are choosing application and features too

b4handjr commented:
Done
<img width="2002" height="771" alt="image" src="https://github.com/user-attachments/assets/442a74ae-82d9-42a9-b081-2f7cdfadc51d" />

yashikakhurana commented on experimenter/experimenter/nimbus_ui/forms.py:
```diff
@@ -1464,21 +1465,26 @@ class FeaturesForm(forms.ModelForm):
-                if application in qs
-            ],
-            key=lambda choice: choice[1].lower(),
+        choices = [("", "Nothing Selected")]  # Add a default blank field.
+        choices.extend(
```
I think you can skip this setting manually `Nothing selected` option, you can refer to locales, countries and channels etc fields they show nothing selected option by default

yashikakhurana commented on experimenter/experimenter/nimbus_ui/templates/nimbus_experiments/features.html:
```diff
@@ -17,24 +17,66 @@
+                </tr>
+              </thead>
+              <tbody>
+                {% for experiment in experiments %}
+                  <tr scope="row">
+                    <td id="delivery-name">
+                      <a target="_blank"
+                         href="{% url 'nimbus-ui-detail' slug=experiment.slug %}"
+                         class="text-decoration-none fw-medium">{{ experiment.name }}</a>
+                    </td>
+                    <td id="delivery-published-date">{{ experiment.published_date|format_not_set }}</td>
+                    <td id="delivery-type-{{ experiment.home_type_choice }}">{{ experiment.home_type_choice }}</td>
+                    <td id="delivery-channel">{{ experiment.get_channel_display }}</td>
+                    <td id="delivery-min-version">{{ experiment.firefox_min_version|parse_version }}</td>
+                    <td id="delivery-population-percent">{{ experiment.population_percent|floatformat:"-1" }}</td>
+                    <td id="delivery-brief">{{ experiment.delivery_brief|format_not_set }}</td>
+                  {% endfor %}
+                </tr>
```
Here can you add a check if no experiment is found matching the desktop and feature configuration to show No deliveries found message

yashikakhurana reviewed (changes requested):
Thank you @b4handjr it is all coming up together, UI is looking really great, found 2 issues, first if I select the application and feature which have experiments available, it shows in the table, but if I change the application it is not updating the deliveries table, you would need to use `htmx-target` and such as to achieve that, second thing if we dont have experiments available having a message will help such as `No deliveries found` 

https://github.com/user-attachments/assets/abb3b75f-845b-494b-b853-606b7415a8e2

b4handjr commented:
Added "No Deliveries" text, I added some padding so it stands out more
<img width="2002" height="816" alt="image" src="https://github.com/user-attachments/assets/5a16495c-27a0-4f8f-9ae7-8bf5d3720dc4" />

yashikakhurana reviewed (changes requested):
ok

okay tested locally and it seems like we cant set ultil we manually set Nothing selected, @b4handjr can you add back nothing selected option and the bug that I see that because it is already selected and not showing the table accordingly until I select it again https://github.com/user-attachments/assets/49b931cc-be8c-4b28-b72c-86720edd489f
option 1- add nothing selected manually
option 2- dont even select the application, then it will not show the features default selected

yashikakhurana reviewed (approved):
perfect, thank you @b4handjr
